woocommerce/compatibility/class-sv-wc-order-compatibility.php

```diff
@@ -522,6 +522,117 @@ public static function get_edit_order_url( \WC_Order $order ) {
+	 *
+	 * @return mixed
+	 */
+	public static function get_order_meta( int $order_id, string $key, $single = null) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+		$value = false;
+
+		// setting $single to false by default if hpos is disabled to match it with get_post_meta method's default value
+		if( null === $single ) {
+			$single = $is_hpos_enabled;
+		}
+
+		if ( $is_hpos_enabled ) {
+			if( $order = wc_get_order( $order_id ) ){
+				$value = $order->get_meta( $key, $single );
+			}
+		} else {
+			$value = get_post_meta( $order_id, $key, $single );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Updates the order meta according to HPOS availability.
+	 *
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional update_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+	 * @param mixed $value Required. Meta value.
+	 */
+	public static function update_order_meta( int $order_id, string $key, $value ) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+
+		if ( $is_hpos_enabled ) {
+			if( $order = wc_get_order( $order_id ) ){
+				$order->update_meta_data( $key, $value, $order_id );
+				$order->save();
+			}
+		} else {
+			update_post_meta( $order_id, $key, $value );
+		}
+	}
+
+
+	/**
+	 * Adds the order meta according to HPOS availability.
+	 *
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional add_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+	 * @param mixed $value Required. meta value.
+	 * @param bool $unique Optional. Whether the same key should not be added. Defaults to false.
+	 */
+	public static function add_order_meta( int $order_id, string $key, $value, $unique = false ) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+
+		if ( $is_hpos_enabled ) {
+			if( $order = wc_get_order( $order_id ) ){
+				$order->add_meta_data( $key, $value, $unique );
+				$order->save();
+			}
+		} else {
+			add_post_meta( $order_id, $key, $value, $unique );
+		}
+	}
+
+	/**
+	 * Deletes the order meta according to HPOS availability.
+	 *
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional delete_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+	 * @param mixed $meta_value Optional. Applicable only if the HPOS is inactive.
+	 */
+	public static function delete_order_meta( int $order_id, string $key ) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+
+		if ( $is_hpos_enabled ) {
+			if( $order = wc_get_order( $order_id ) ){
+				$order->delete_meta_data( $key );
+				$order->save();
+			}
+		} else {
+			delete_post_meta( $order_id, $key, $meta_value );
```

Comment:
delete_post_meta does not require `$meta_value`  - this variable is not present and would cause a fatal error

Reply:
@unfulvio-godaddy according to this function [reference here](https://developer.wordpress.org/reference/functions/delete_post_meta/), it accepts the third $meta_value parameter which is by default set to an empty string as I did in my method, but yes I don't think we're using this third parameter anywhere, but I added it to keep it same as core method. 

Let me know if I should remove it?

Reply:
I almost forgot that function supported a 3rd arg actually. The variable wasn't available in the method's logic when I first reviewed. Anyhow, sounds like a bit like https://github.com/godaddy-wordpress/wc-plugin-framework/pull/581#discussion_r1066512178 -- I doubt we use delete_post_meta in our codebase with ever passing the 3rd arg - sounds good leaving the default value as you did; this will have no effect with HPOS on so I don't know if we need it at all. Fine for me to leave as is.